### PR TITLE
Add VSLAM-demo example

### PR DESCRIPTION
**Pull request by jike5**
What type of PR is this?

https://github.com/kubeedge/examples/issues/120

https://github.com/kubeedge/community/pull/92

What this PR does / why we need it:

[OSPP] Code for Implementation of an VSLAM Algorithm on KubeEdge

In order to develop a practical system, we follow the architectural design from [edgeslam](https://github.com/droneslab/edgeslam), still, with numerous modifications for our scenario. Relying on the cloud-edge management and cloud-edge communication function provided by kubeedge and edgemesh, a complete cloud-edge collaborative SLAM system is implemented. In this system, the edge side is responsible for sensor data collection, feature extraction and data sending, while the cloud side is in charge of loop closing and global mapping.

### CODE REVIEW

**Comment by JoeyHwong-gk** [comment]
Thanks for your contribution!

As the `proposal` is repeated with the previous PR #92 , just keep one of them.
--
Rebase commits when your work is done.
---
Push your container image  to dockerhub and submit more source code please.
---
Please add the license to the top of the execution scripts

File: sig-robotics/examples/VSLAM-demo/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   ```bash
+   docker image pull jike5/orbslam2-dev:latest
+   ```
+
+4. Build slam applications image based on `orbslam2-dev` image
+
+   ```
+   docker build . -t jike5/edgeslam-run -f docker/edgeslam-run/Dockerfile
+   ```
+
```
[the remark this answers is not in the document]
Comment (jike5) [reply sig-robotics/examples/VSLAM-demo/README.md:31]: No problem, actually i have push the image on dockerhub already.

File: sig-robotics/examples/VSLAM-demo/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   ```bash
+   kubectl apply -f edge-deployment.yaml
+   kubectl apply -f cloud-deployment.yaml
+   ```
+
+8. Get inside the container
+
+   ```bash
+   docker ps -a | grep orbslam
+   docker exec -it [containerID] bash
```
[the remark this answers is not in the document]
Comment (jike5) [reply sig-robotics/examples/VSLAM-demo/README.md:80]: ok, you are right

File: sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+   git clone https://github.com/kubeedge/community
+   ```
+
+3. Prepare your RGBD dataset or RGBD-sensors on the edge-side.
+
+   > We use [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) and its format in our development and test process.
+
+4. Check your work node's name and modify the deployment.yaml file.
+
+   > For example, we have a master node and two work nodes, named edge01 and edge02
```
Comment (JoeyHwong-gk) [inline sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md:26]: `master node` -> `cloud node`, `work nodes` -> `edge nodes`

File: sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+   `edge-deployment.yaml`
+
+   ```yaml
+   spec:
+     template:
+       spec:
+         nodeName: edge01
+   ```
+
+   Check the location of the downloaded  [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) and modify `edge-deployment.yaml`
```
Comment (JoeyHwong-gk) [inline sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md:46]: download [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) to your edge node, and mount it within k8s.

File: sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+   ```bash
+   kubectl apply -f edge-deployment.yaml
+   kubectl apply -f cloud-deployment.yaml
+   ```
+
+7. If your edge-side node support X11 GUI, you will see a frame windows and a map window. The trajectory will be saved under `/workspace/edgeslam/Examples`  folder.
+
+## Validation
+
+We tested origin [ORB_SLAM2]() and ours on the  [TUM datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download).
```
Comment (JoeyHwong-gk) [inline sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md:84]: add reference link to ORB_SLAM2

Comment (JoeyHwong-gk) [inline sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md:84]: The works we has use for reference including [ORB_SLAM2](https://github.com/raulmur/ORB_SLAM2), and combine the performance on [TUM datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download). We illustrate the overall framework of ours in the Figure:

File: sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/build/edgeslam-run/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
+FROM   jike5/orbslam2-dev:latest
```
Comment (JoeyHwong-gk) [inline sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/build/edgeslam-run/Dockerfile:1]: an extra space

File: sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+   ```bash
+   git clone https://github.com/kubeedge/community
+   ```
+
+3. Prepare your RGBD dataset or RGBD-sensors on the edge-side.
+
+   > We use [TUM Datasets](https://vision.in.tum.de/data/datasets/rgbd-dataset/download) and its format in our development and test process.
+
+4. Check your work node's name and modify the deployment.yaml file.
```
[the remark this answers is not in the document]
Comment (jike5) [reply sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md:24]: Directly replace the sentence `4. Check your work...` with this one?

[the remark this answers is not in the document]
Comment (jike5) [reply sig-robotics/examples/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md]: i have modified it, thanks a lot

**Pull request merged**